src/slack/utils.bal

```diff
@@ -0,0 +1,436 @@
+                url + INITIAL_COMMENT_AS_FIRST_PARAM + initialComment;
+    }
+    if (threadTs is string) {
+        url = (stringutils:contains(url, QUESTION_MARK)) ? url + THREAD_TS_AS_SECOND_PARAM + threadTs : 
+                url + THREAD_TS_AS_FIRST_PARAM + threadTs;
+    }              
+    http:Request request = new;
+    mime:Entity filePart = new;
+    filePart.setContentDisposition(getContentDispositionForFormData(FILE, filePath));
+
+    filePart.setFileAsEntityBody(filePath);
+    mime:Entity[] bodyParts = [<@untainted> filePart];
+    request.setBodyParts(bodyParts, contentType = mime:MULTIPART_FORM_DATA);
+    http:Response|error response = slackClient->post(url, request);
+    if (response is error) {
+        return setResError(response);
+    }
+    http:Response resp = <http:Response> response;
+    var fileInfo = resp.getJsonPayload();
+    if (fileInfo is error) {
+        return setJsonResError(fileInfo);
+    }
+    json fileInfoPayload = <json> fileInfo;  
+    var checkOk = check checkOk(fileInfoPayload);    
+    var file = check fileInfoPayload.file;
+    return FileInfo.constructFrom(file);
+}
+
+function getFileInfo(http:Client slackClient, string fileId) returns @tainted FileInfo|error {
+    string url = GET_FILE_INFO_PATH + fileId;
+    http:Response|error response = slackClient->get(url);
+    if (response is error) {
+        return setResError(response);
+    }
+    http:Response resp = <http:Response> response;
+    var fileInfo = resp.getJsonPayload();
+    if (fileInfo is error) {
+        return setJsonResError(fileInfo);
+    }
+    json fileInfoPayload = <json> fileInfo;
+    var checkOk = check checkOk(fileInfoPayload);
+    var file = check fileInfoPayload.file;
+    return FileInfo.constructFrom(file);
+}
+
+function getContentDispositionForFormData(string partName, string filePath) returns (mime:ContentDisposition) {
+    mime:ContentDisposition contentDisposition = new;
+    contentDisposition.name = partName;
+    contentDisposition.disposition = DISPOSITION;
+    contentDisposition.fileName = getFileName(filePath);
+    return contentDisposition;
+}
+
+function getFileName(string filePath) returns string {
+    int lastIndex = stringutils:lastIndexOf(filePath, BACK_SLASH);
+    return filePath.substring(lastIndex + 1);
+}
+
+function handleOkResp(http:Client slackClient, string url) returns @tainted error? {
+    http:Response|error response = slackClient->post(url, EMPTY_STRING);
+    if (response is error) {
+        return setResError(response);
+    }
+    http:Response resp = <http:Response> response;
+    var jsonResp = resp.getJsonPayload();
+    if (jsonResp is error) {
+        return setJsonResError(jsonResp);
+    }
+    json jsonPayload = <json> jsonResp;
+    var checkOkResp = checkOk(jsonPayload);
+}
+
+function setResError(error errorResponse) returns error {
+    return error(SLACK_ERROR_CODE, message = <string> errorResponse.detail()?.message);
```

Comment:
Shall we set the `errorResponse` as the cause of the error, and set the message to Slack connector specific one?
something like:
```suggestion
    return error(SLACK_ERROR_CODE, message = "Error received from the slack server", cause = errorResponse);
```